java/vector/src/test/java/org/apache/arrow/vector/TestValueVector.java

```diff
@@ -1394,6 +1394,35 @@ public void testFillEmptiesNotOverfill() {
     }
   }
 
+  @Test
+  public void testSetSafeWithArrowBuf() {
+    final int numValues = BaseFixedWidthVector.INITIAL_VALUE_ALLOCATION * 2;
+
+    try (
+        final VarCharVector fromVector = newVector(VarCharVector.class, EMPTY_SCHEMA_PATH,
+            MinorType.VARCHAR, allocator);
+        final VarCharVector toVector = newVector(VarCharVector.class, EMPTY_SCHEMA_PATH,
+            MinorType.VARCHAR, allocator)) {
+      fromVector.setInitialCapacity(numValues);
```

Comment:
it might make the unit test easier to read if you separated each section by a comment.
i.e.
//Setup

// Execute

// Verify

Reply:
i've added comments.